daffodil-test/src/test/scala-debug/org/apache/daffodil/section13/text_number_props/TestTextNumberPropsDebug.scala

```diff
@@ -34,34 +34,22 @@ class TestTextNumberPropsDebug {
 
   import TestTextNumberPropsDebug._
 
-  // DFDL-845
-  @Test def test_textNumberCheckPolicy_lax01() { runner.runOneTest("textNumberCheckPolicy_lax01") }
-  @Test def test_textNumberCheckPolicy_lax05() { runner.runOneTest("textNumberCheckPolicy_lax05") }
-  @Test def test_textNumberCheckPolicy_lax04() { runner.runOneTest("textNumberCheckPolicy_lax04") }
-  @Test def test_textNumberCheckPolicy_lax10() { runner.runOneTest("textNumberCheckPolicy_lax10") }
-  @Test def test_textNumberCheckPolicy_lax11() { runner.runOneTest("textNumberCheckPolicy_lax11") }
-  @Test def test_textNumberCheckPolicy_lax12() { runner.runOneTest("textNumberCheckPolicy_lax12") }
-  @Test def test_textNumberCheckPolicy_lax13() { runner.runOneTest("textNumberCheckPolicy_lax13") }
-  @Test def test_textNumberCheckPolicy_lax14() { runner.runOneTest("textNumberCheckPolicy_lax14") }
-  @Test def test_textNumberCheckPolicy_lax15() { runner.runOneTest("textNumberCheckPolicy_lax15") }
-  @Test def test_textNumberCheckPolicy_lax16() { runner.runOneTest("textNumberCheckPolicy_lax16") }
-
   // DFDL-847
   @Test def test_textStandardDecimalSeparator10() { runner.runOneTest("textStandardDecimalSeparator10") }
   @Test def test_textStandardDecimalSeparator11() { runner.runOneTest("textStandardDecimalSeparator11") }
 
-  // DFDL-851
-  @Test def test_textStandardGroupingSeparator12() { runner.runOneTest("textStandardGroupingSeparator12") }
-  @Test def test_textStandardDecimalSeparator17() { runner.runOneTest("textStandardDecimalSeparator17") }
-  @Test def test_standardZeroRep07() { runner.runOneTest("standardZeroRep07") }
-
   // DFDL-853
   @Test def test_textNumberPattern_pSymbol01() { runner.runOneTest("textNumberPattern_pSymbol01") }
   @Test def test_textNumberPattern_pSymbol02() { runner.runOneTest("textNumberPattern_pSymbol02") }
 
-  @Test def test_textNumberPattern_scientificNotation02() { runner.runOneTest("textNumberPattern_scientificNotation02") }
-
-  //DFDL-191
+  // DFDL-861
   @Test def test_infnanCaseInsensitive() { runner.runOneTest("infnanCaseInsensitive") }
   @Test def test_expCaseSensitive() { runner.runOneTest("expCaseSensitive") }
+
+  // DAFFODIL-XXXX
+  @Test def test_expEmptyString() { runner.runOneTest("expEmptyString") }
+  @Test def test_expEmptyString2() { runner.runOneTest("expEmptyString2") }
+
+  // DAFFODIL-XXXX
+  @Test def test_textNumberPattern_padding09() { runner.runOneTest("textNumberPattern_padding09") }
```

Comment:
When adding a test for the bug in ICU that caused this test to fail, I discovered the actual problem is that ICU does not update the ParsePosition when there is suffix padding. For example, if the string was "123 SUFFIX______", and the text number pattern defines an after-suffix pad character of _, the ParsePosition points to "X" rather than the end of the string. This means that we can manually check to see if all character after ParsePosition are the pad character if the pad position is after suffix to see if a string is valid, and thus maintain the original behavior. Ill still add a test to see if ICU ever fixes this bug, but this test no longer fails with the mentioned fix.